### PR TITLE
bugfix: fix when TC returns RollbackFailed FailureHandler not executed

### PR DESCRIPTION
**Pull request by ZhangShiYeChina**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
fixes :https://github.com/seata/seata/issues/5231

image

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

File: tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java [hunk 1]
```diff
@@ -239,9 +239,29 @@ private void rollbackTransaction(GlobalTransaction tx, Throwable originalExcepti
...
+            case RollbackRetrying:
+                code = TransactionalExecutor.Code.RollbackRetrying;
+                break;
+            case TimeoutRollbacking:
+            case TimeoutRollbackRetrying:
+            case TimeoutRollbacked:
+                code = TransactionalExecutor.Code.TimeoutRollback;
+                break;
+            case Rollbacked:
+            default:
```
Comment (Bughue) [inline tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java:260]: 我觉得只有Rollbacked是RollbackDone吧？如果真的到default了是不是该抛个其他异常了？

Comment (slievrly) [reply tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java:260]: Consider the compatibility of high and low versions (client is a low version, server is a high version and the new state), do not throw exceptions.You can just print a log.

File: tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java [hunk 2]
```diff
@@ -239,9 +239,29 @@ private void rollbackTransaction(GlobalTransaction tx, Throwable originalExcepti
...
+            case RollbackFailed:
+            case TimeoutRollbackFailed:
+            case RollbackRetryTimeout:
+                code = TransactionalExecutor.Code.RollbackFailure;
+                break;
+            case Rollbacking:
+            case RollbackRetrying:
+                code = TransactionalExecutor.Code.RollbackRetrying;
+                break;
+            case TimeoutRollbacking:
```
Comment (slievrly) [inline tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java:254]: The next  pr will removes the TimeRollback state of the client

**Pull request merged**